velox/docs/develop/types.rst

```diff
@@ -2,54 +2,125 @@
-
-======================  ===========================    ==================
-Velox Type              C++ Type                       Bytes per Value
-======================  ===========================    ==================
-BOOLEAN                 bool                            0.125 (i.e. 1 bit)
-TINYINT                 int8_t                          1
-SMALLINT                int16_t                         2
-INTEGER                 int32_t	                        4
-BIGINT                  int64_t                         8
-DATE                    struct Date                     8
-REAL                    float                           4
-DOUBLE                  double                          8
-SHORT_DECIMAL           struct UnscaledShortDecimal     8
-LONG_DECIMAL            struct UnscaledLongDecimal     16
-TIMESTAMP               struct Timestamp               16
-INTERVAL DAY TO SECOND  struct IntervalDayTime          8
-VARCHAR                 struct StringView              16
-VARBINARY               struct StringView              16
-======================  ===========================    ==================
-
-All scalar types except DECIMAL have a one-to-one mapping with their C++ types.
-
-DECIMAL type is a special scalar type since it carries additional `precision`,
-and `scale` information. Similar to SQL decimal, `precision` is the number of
+Physical Types
+~~~~~~~~~~~~~~
+Each physical type is implemented using a C++ type. The table
+below shows the supported physical types, their corresponding C++ type,
+and fixed-width bytes required per value.
+
+================   ===========================   ===================
+Physical Type      C++ Type                      Fixed Width (bytes)
+================   ===========================   ===================
+BOOLEAN            bool                          0.125 (i.e. 1 bit)
+TINYINT            int8_t                        1
+SMALLINT           int16_t                       2
+INTEGER            int32_t	                     4
+BIGINT             int64_t                       8
+HUGEINT            int128_t                      16
+DATE               struct Date                   8
+REAL               float                         4
+DOUBLE             double                        8
+TIMESTAMP          struct Timestamp              16
+VARCHAR            struct StringView             16
+VARBINARY          struct StringView             16
+OPAQUE             std::shared_ptr<void>         16
+UNKNOWN            struct UnknownValue           0
+================   ===========================   ===================
+
+All physical types except VARCHAR and VARBINARY have a one-to-one mapping
+with their C++ types.
+The C++ type is also used as a template parameter for vector classes.
+For example, vector of 64-bit integers is represented as `FlatVector<int64_t>`
+whose type is `BIGINT`.
+
+OPAQUE type can be used to define custom types.
```

Comment:
Can we move discussion of custom types after Logical Types section?

Reply:
There is a  Custom Types section after Logical Types that has all the details.
I am only introducing OPAQUE type here since we discuss variable-width types below.